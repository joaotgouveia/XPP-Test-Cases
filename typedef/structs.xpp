type my_type = struct { a: int, b: int };
(
let f = fn x: my_type => { x }; f({ a = 1; b = 2 })
);;

type my_type = struct { a: struct { b: int }, c: int };
(
let f = fn x: my_type => { x }; f({ a = { b = 1 }; c = 2 })
);;
