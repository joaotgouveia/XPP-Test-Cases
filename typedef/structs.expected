Value: record { a = 1; b = 2 }
Type: record { a: int; b: int }

Value: record { a = record { b = 1 }; c = 2 }
Type: record { a: record { b: int }; c: int }
